cn_autograd_debugging/PopulationData_ag.py

```diff
@@ -0,0 +1,313 @@
+
+"""
+PopulationData.py
+
+--------
+
+"""
+
+import os
+import turicreate as tc
+import autograd.numpy as np
+from datetime import datetime,timedelta
+from pathlib import Path
+
+# from scipy.stats import gamma
+# import autograd.scipy.stats.gamma as gamma
+# import autograd.scipy.stats as stat
+
+from autograd.scipy.special import gamma as gamma_fcn
+from autograd.scipy.stats import beta,norm
+import autograd
+
+import csv
+import requests
+from bs4 import BeautifulSoup
+import urllib
+
+
+
+
+
+
+def log_gamma_pdf(x,alpha,beta):
+    return np.log((beta**alpha)*(x**(alpha-1))*np.exp(-beta*x)/gamma_fcn(alpha))
```

Comment:
should do this all in log space directly, rather than compute pdf and then take the log

Reply:
You mean using the log rules of multiplication and division? such that the resulting return value is a series of logs?

[alpha*np.log(beta)] + [(alpha-1)*np.log(x)] - [beta*x]   - np.log(gamma_fcn(alpha))

Is the reason just for numerical stability with extreme values?